agent/structs/config_entry.go

```diff
@@ -82,6 +82,14 @@ type UpdatableConfigEntry interface {
 	ConfigEntry
 }
 
+// WarningConfigEntry is an optional interface implemented by a ConfigEntry
+// if it wants to be able to emit warnings when it is being upserted.
+type WarningConfigEntry interface {
+	Warnings() []string
```

Comment:
I was thinking of making this part of the values returned by `Validate()` but that seemed like overkill for just this.

Reply:
I personally like the approach you took optional interface implementation and type assertions.